Purchases/Networking/HTTPStatusCode.swift

```diff
@@ -14,14 +14,73 @@
 
 import Foundation
 
-enum HTTPStatusCode: Int {
-
-    case success = 200,
-         createdSuccess = 201,
-         redirect = 300,
-         notModifiedResponseCode = 304,
-         internalServerError = 500,
-         notFoundError = 404,
-         networkConnectTimeoutError = 599
+enum HTTPStatusCode {
+
+    case success
+    case createdSuccess
+    case redirect
+    case notModified
+    case invalidRequest
+    case notFoundError
+    case internalServerError
+    case networkConnectTimeoutError
+
+    case other(Int)
+
+    private static let knownStatus: Set<HTTPStatusCode> = [
+        .success,
+        .createdSuccess,
+        .redirect,
+        .notModified,
+        .invalidRequest,
+        .notFoundError,
+        .internalServerError,
+        .networkConnectTimeoutError
+    ]
+    private static let statusByCode: [Int: HTTPStatusCode] = Self.knownStatus.dictionaryWithKeys { $0.rawValue }
+}
+
+extension HTTPStatusCode: RawRepresentable {
+
+    init(rawValue: Int) {
+        self = Self.statusByCode[rawValue] ?? .other(rawValue)
+    }
+
+    var rawValue: Int {
+        switch self {
+        case .success: return 200
+        case .createdSuccess: return 201
+        case .redirect: return 300
+        case .notModified: return 304
+        case .invalidRequest: return 400
+        case .notFoundError: return 404
+        case .internalServerError: return 500
+        case .networkConnectTimeoutError: return 599
+
+        case let .other(code): return code
+        }
+    }
+
+}
+
+extension HTTPStatusCode: ExpressibleByIntegerLiteral {
+
+    init(integerLiteral value: IntegerLiteralType) {
+        self.init(rawValue: value)
+    }
+
+}
+
+extension HTTPStatusCode: Hashable {}
+
+extension HTTPStatusCode {
+
+    var isSuccessfulResponse: Bool {
+        return self.rawValue < HTTPStatusCode.redirect.rawValue
+    }
```

Comment:
There _are_ some HTTP codes in the 100s. Something like the following feels a little 
```suggestion
    var isSuccessfulResponse: Bool {
        return 200...299 ~= self.rawValue
    }
```

It feels weird to not use our status enums here but HTTP codes seems okay to do something like this? 

Totally fine if we don't like what I suggested but it feels a little easier to interpret, IMO 😇

Reply:
I kept the existing logic, but I considered this. I'm glad you brought it up, I think this is better moving forward 👍🏻